### DISCUSSION TITLE
Production bug

### DISCUSSION THREAD

**Discussion by sourabhmahato** [nextauthjs/next-auth#2506] [Help]
### Description 🐜

![image](https://user-images.githubusercontent.com/54588817/128682749-1065fe59-7b65-4443-a605-dc252f08557b.png)
After singing in through credentials when ever I go back to /login client side, I get this, and when clear site data I can see sign page. But I should see signout button instead this image.  

### Is this a bug in your own project?

Yes

### How to reproduce ☕️

```js
function login() {
  const [session, loading] = useSession()

  
  if (loading) return <div>Loading</div>
  return (
    <div>
      <main>
        {!session && (
          <>
            <div style={{ textAlign: "center", height: "500px" }}>
              <h1>Please, Sign In!</h1> <br />
              <button onClick={() => signIn()}>Sign In</button>
            </div>
          </>
        )}
        {session && (
          <>
            <div style={{ textAlign: "center", height: "500px" }}>
              <h1>Signed In as {session.user.name}</h1> <br />
              {session.user.name === "adminHR" ? (
                <>
                 
                  <div>
                    <h3 style={{ marginBottom: "20px" }}>
                      You can access authorized admin pages
                    </h3>
                  </div>
                  <button>
                    <Link href="/database/data-display">
                      To our Career Dashboard
                    </Link>
                  </button>
                </>
              ) : (
                <></>
              )}{" "}
              {session.user.name === "adminRecipe" ? (
                <>
                  <div>
                    <h3 style={{ marginBottom: "20px" }}>
                      You can access authorized admin pages
                    </h3>
                  </div>
                  <button>
                    <Link href="/dashboard">To our Recipe Dashboard</Link>
                  </button>
                </>
              ) : (
                <></>
              )}
              {session.user.name !== "adminHR" &&
              session.user.name !== "adminRecipe" ? (
                <>
                  <div>
                    {" "}
                    <h2 style={{ marginBottom: "20px" }}>
                      dont have access to any routes with this credentials
                    </h2>
                  </div>
                </>
              ) : (
                <></>
              )}
              <button style={{ marginLeft: "10px" }} onClick={() => signOut()}>
                Sign Out
              </button>
            </div>
          </>
        )}
      </main>
    </div>
  )
}

export default login
```

client side code for signin and signout

### Environment 🖥

all file are in .env

### Contributing 🙌🏽

Yes,  I am willing to help solve this bug in a PR

**Comment by sourabhmahato** [comment]
Please someone help I am not able solve this issue

**Reply by balazsorban44** [reply]
Please provide a proper reproduction. Unfortunately the code you pasted in is way too generic for us to be able to give you more guidance.

Do you have some logs at least?

It is very likely not an issue with `next-auth`

**Reply by sourabhmahato** [reply]
(node:18) UnhandledPromiseRejectionWarning: SyntaxError: Unexpected token o in JSON at position 1
    at JSON.parse (<anonymous>)
    at Object.encode (/app/node_modules/next-auth/dist/lib/jwt.js:40:65)
    at Object.callback (/app/node_modules/next-auth/dist/server/routes/callback.js:318:37)
    at processTicksAndRejections (internal/process/task_queues.js:95:5)
(Use `node --trace-warnings ...` to show where the warning was created)
(node:18) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 17)
(node:18) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.

this is the log error

**Reply by balazsorban44** [reply]
could you also show your `[...nextauth].js` file?

Even better, a reproduction repository?
